gammapy/analysis/core.py

```diff
@@ -411,20 +442,20 @@ def _spectrum_extraction(self):
         reference = SpectrumDataset.create(geom=geom, energy_axis_true=e_true)
 
         datasets = []
-        for obs in self.observations:
-            log.info(f"Processing observation {obs.obs_id}")
-            dataset = dataset_maker.run(reference.copy(), obs)
-            if bkg_maker is not None:
-                dataset = bkg_maker.run(dataset, obs)
-                if dataset.counts_off is None:
-                    log.info(
-                        f"No OFF region found for observation {obs.obs_id}. Discarding."
-                    )
-                    continue
-            dataset = safe_mask_maker.run(dataset, obs)
-            log.debug(dataset)
-            datasets.append(dataset)
-
+        with pbar(total=len(self.observations), show_pbar=show_pbar, desc="Datasets") as pb:
+            for obs in self.observations:
+                dataset = dataset_maker.run(reference.copy(), obs)
+                if bkg_maker is not None:
+                    dataset = bkg_maker.run(dataset, obs)
+                    if dataset.counts_off is None:
+                        log.info(
```

Comment:
There are two more options here:
- switch to log.debug()
- log if show_pbar is False and dataset.counts_off is None

We can also keep as it is if you consider that this info is important enough to break the progress bar display